drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/test/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNTypeServiceImplTest.java

```diff
@@ -46,125 +56,185 @@ public void init() {
         // nested collection
-        DMNType person = new CompositeTypeImpl();
-
-        SimpleTypeImpl numbers = new SimpleTypeImpl(null, "string", null);
-        numbers.setCollection(true);
-
-        DMNType phoneNumberCompositeCollection = new CompositeTypeImpl(null, "tPhoneNumber", null, true);
-        ((CompositeTypeImpl) phoneNumberCompositeCollection).addField("numbers", numbers);
-
-        ((CompositeTypeImpl) person).addField("phoneNumbers", phoneNumberCompositeCollection);
-
-        errorHolder = new DMNTypeServiceImpl.ErrorHolder();
-        dmnTypeServiceImpl.checkTypeSupport(person, errorHolder, "fieldName");
+        CompositeTypeImpl singleCompositeWithComplexCollection = getSingleCompositeWithNestedCollection();
+        DMNTypeServiceImpl.ErrorHolder errorHolder = new DMNTypeServiceImpl.ErrorHolder();
+        dmnTypeServiceImpl.checkType(singleCompositeWithComplexCollection, errorHolder, "fieldName");
         assertEquals(0, errorHolder.getTopLevelCollection().size());
         assertEquals(0, errorHolder.getMultipleNestedObject().size());
         assertEquals(1, errorHolder.getMultipleNestedCollection().size());
         assertEquals("fieldName.phoneNumbers.numbers", errorHolder.getMultipleNestedCollection().get(0));
+    }
 
+    @Test
+    public void checkTypeSingleCompositeWithCollection() {
         // nested object into collection
-        person = new CompositeTypeImpl();
-
+        CompositeTypeImpl person = new CompositeTypeImpl();
         CompositeTypeImpl complexNumbers = new CompositeTypeImpl(null, "tPhoneNumber", null, false);
-
-        phoneNumberCompositeCollection = new CompositeTypeImpl(null, "tPhoneNumber", null, true);
-        ((CompositeTypeImpl) phoneNumberCompositeCollection).addField("complexNumbers", complexNumbers);
-
-        ((CompositeTypeImpl) person).addField("phoneNumbers", phoneNumberCompositeCollection);
-
-        errorHolder = new DMNTypeServiceImpl.ErrorHolder();
-        dmnTypeServiceImpl.checkTypeSupport(person, errorHolder, "fieldName");
+        CompositeTypeImpl phoneNumberCompositeCollection = new CompositeTypeImpl(null, "tPhoneNumber", null, true);
+        phoneNumberCompositeCollection.addField("complexNumbers", complexNumbers);
+        person.addField(EXPANDABLE_PROPERTY_PHONENUMBERS, phoneNumberCompositeCollection);
+        DMNTypeServiceImpl.ErrorHolder errorHolder = new DMNTypeServiceImpl.ErrorHolder();
+        dmnTypeServiceImpl.checkType(person, errorHolder, "fieldName");
         assertEquals(0, errorHolder.getTopLevelCollection().size());
         assertEquals(1, errorHolder.getMultipleNestedObject().size());
         assertEquals(0, errorHolder.getMultipleNestedCollection().size());
         assertEquals("fieldName.phoneNumbers.complexNumbers", errorHolder.getMultipleNestedObject().get(0));
     }
+
+    private void commonFactModelTreeValidation(FactModelTuple factModelTuple, DMNNode dmnNode, SortedMap<String, FactModelTree> hiddenFacts) {
```

Comment:
This entire method is a leap of faith for me :D 
Just kidding, it seems fine to me but it is still really hard to follow

Reply:
@danielezonca 
_REFACTORED_